numpy/core/arrayprint.py

```diff
@@ -861,7 +863,8 @@ def fillFormat(self, data):
 
         # account for sign = ' ' by adding one to pad_left
         if all(finite_vals >= 0) and self.sign == ' ':
-            self.pad_left += 1
+            if self._legacy != '1.13':
+                self.pad_left += 1
```

Comment:
Would read better with the `if`s inverted (and comment moved within the inner one), I think